### PR TITLE
Add more messages + conversions

### PR DESCRIPTION
**Pull request by oswinso**
## Description
This PR adds pose, twist and `BlueTeam` messages, as well as conversion to / from them.

### CODE REVIEW

File: rj_msgs/msg/RobotState.msg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This message contains data that comes from the vision system
+# including position data and which camera this robot was seen by and
+# what time it was last seen.
+# TODO(?): Revisit the "which camera" part. It seems to be missing in WorldState.hpp.
```
Comment (JNeiger) [inline rj_msgs/msg/RobotState.msg]: If this is the output of the vision filter, it shouldn't have the camera, just pose/time/visible

**Review by JNeiger — APPROVED** [review]

File: rj_msgs/msg/RobotState.msg (hunk 1 above)
Comment (oswinso) [reply rj_msgs/msg/RobotState.msg]: I'm taking this comment from `RobotState.hpp`, which does include the "which camera" part in the comments:
https://github.com/RoboJackets/robocup-software/blob/fb55e0963d021e8625778f667260addb33dc9031/soccer/WorldState.hpp#L12-L23

Comment (JNeiger) [reply rj_msgs/msg/RobotState.msg]: @kylestach What was the thought behind this?

**Comment by kylestach** [comment]
`GameState` should not be a message at this time. With referee, I was planning to split it into four things:
 - Team color
 - Goalie ID (technically replicated from team info, but it's more important at the system level)
 - Team info (red cards, duplicated goalie ID, yellow cards/timing, team name, etc.)
 - Game info (period, state, current restart information, time left in current stage)
Pose and twist are fine.

**Comment by oswinso** [comment]
> `GameState` should not be a message at this time. With referee, I was planning to split it into four things:
> 
>     * Team color
> 
>     * Goalie ID (technically replicated from team info, but it's more important at the system level)
> 
>     * Team info (red cards, duplicated goalie ID, yellow cards/timing, team name, etc.)
> 
>     * Game info (period, state, current restart information, time left in current stage)
>       Pose and twist are fine.

What do you mean by "should not be a message at this time"?

**Comment by kylestach** [comment]
You don't need to change that because it's being changed in referee. But also, it isn't a single atomic message.

**Comment by oswinso** [comment]
> You don't need to change that because it's being changed in referee. But also, it isn't a single atomic message.

Do you mind making a PR then? I need this message for `VisionReciever`. More specifically, I need `blue_team`.

**Comment by kylestach** [comment]
I have `blue_team` as its own message, you can just add that one.

File: rj_msgs/msg/RobotState.msg (hunk 1 above)
Comment (kylestach) [reply rj_msgs/msg/RobotState.msg]: Uh yeah that doesn't sound right. Probably an old comment that didn't get updated, rip.

File: rj_constants/include/rj_constants/topic_names.hpp [hunk 2]
```diff
@@ -7,12 +7,19 @@
 namespace config_server::topics {
 constexpr auto kGameSettingsPub = "config/game_settings";
 constexpr auto kFieldDimensionsPub = "config/field_dimensions";
+constexpr auto kGameStatePub = "config/game_state";
```
Comment (kylestach) [inline rj_constants/include/rj_constants/topic_names.hpp]: Remove

File: Geometry2d/include/Geometry2d/Pose.hpp [hunk 3]
```diff
@@ -1,6 +1,9 @@
 #pragma once
 
 #include <ostream>
+#include <rj_geometry_msgs/msg/detail/pose__builder.hpp>
+#include <rj_geometry_msgs/msg/detail/twist__builder.hpp>
```
Comment (kylestach) [inline Geometry2d/include/Geometry2d/Pose.hpp]: Do the regular pose.hpp/twist.hpp not include the builders?

File: soccer/src/soccer/WorldState.hpp [hunk 4]
```diff
@@ -17,16 +20,52 @@ constexpr double kBallDecayConstant = 0.180;
  * what time it was last seen.
  */
 struct RobotState {
+    using Msg = rj_msgs::msg::RobotState;
```
Comment (kylestach) [inline soccer/src/soccer/WorldState.hpp:23]: I think we should standardize exactly how we want to keep parity between C++/ROS/[protobuf, optionally] types and keep it consistent. If we just want to do a bunch of implicit conversions and an associated `Msg` type I _think_ that's fine, (although template magic would make things a little bit easier for i.e. autogenerating specializations for a vector of ros types).

File: rj_geometry_msgs/msg/Pose.msg [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Represents a pose in 2d space: (x, y, theta). Uses double-precision floating
+# point numbers for the coordinates, and can be used to represent the position
+# and orientation of a robot in the plane.
+
+Point position
+float64 heading
```
Comment (kylestach) [inline rj_geometry_msgs/msg/Pose.msg]: Newline

File: rj_param_utils/include/rj_param_utils/param.h [hunk 6]
```diff
@@ -180,13 +180,13 @@ class ParamProvider {
  * @param description A description of the parameter.
  */
 #define DEFINE_VARIABLE(type, module, name, val, description)              \
-    namespace params::variables {                                          \
+    namespace params::storage {                                            \
```
Comment (kylestach) [inline rj_param_utils/include/rj_param_utils/param.h]: Didn't this stuff get merged in in #1550?

File: soccer/src/soccer/WorldState.hpp [hunk 7]
```diff
@@ -3,6 +3,9 @@
 #include <Geometry2d/Pose.hpp>
 #include <rj_common/time.hpp>
 #include <rj_constants/constants.hpp>
+#include <rj_msgs/msg/detail/ball_state__builder.hpp>
+#include <rj_msgs/msg/detail/robot_state__builder.hpp>
+#include <rj_msgs/msg/detail/world_state__builder.hpp>
```
Comment (kylestach) [inline soccer/src/soccer/WorldState.hpp]: Include regular message headers instead of __builders.

File: soccer/src/soccer/WorldState.hpp [hunk 8]
```diff
@@ -17,16 +20,52 @@ constexpr double kBallDecayConstant = 0.180;
...
+    /**
+     * @brief Implicit conversion to RobotState::Msg.
+     * @return
+     */
+    [[nodiscard]] operator Msg() const {
+        return rj_msgs::build<Msg>()
+            .stamp(RJ::ToROSTime(timestamp))
+            .pose(pose)
+            .velocity(velocity)
+            .visible(visible);
```
Comment (kylestach) [inline soccer/src/soccer/WorldState.hpp:59]: We should probably have unit tests to ensure that the `T -> T::Msg -> T` cycle is a nop.

**Review by kylestach — APPROVED** [review]

File: Geometry2d/include/Geometry2d/Pose.hpp (hunk 3 above)
Comment (oswinso) [reply Geometry2d/include/Geometry2d/Pose.hpp]: nope

File: rj_param_utils/include/rj_param_utils/param.h (hunk 6 above)
Comment (oswinso) [reply rj_param_utils/include/rj_param_utils/param.h]: that's true, this change is accidental.

Comment (oswinso) [reply rj_param_utils/include/rj_param_utils/param.h]: actually idk why it shows this stuff as being changed. I don't see these changes when doing a `git diff ros2`

File: Geometry2d/include/Geometry2d/Pose.hpp (hunk 3 above)
Comment (oswinso) [reply Geometry2d/include/Geometry2d/Pose.hpp]: nvm they do

File: soccer/src/soccer/WorldState.hpp (hunk 7 above)
Comment (oswinso) [reply soccer/src/soccer/WorldState.hpp]: done.

File: rj_constants/include/rj_constants/topic_names.hpp (hunk 2 above)
Comment (oswinso) [reply rj_constants/include/rj_constants/topic_names.hpp]: Done.

File: rj_geometry_msgs/msg/Pose.msg (hunk 5 above)
Comment (oswinso) [reply rj_geometry_msgs/msg/Pose.msg]: Done.

File: soccer/src/soccer/WorldState.hpp (hunk 8 above)
Comment (oswinso) [reply soccer/src/soccer/WorldState.hpp:59]: added.

**Pull request merged**